src/components/NewDesign/Swap/Swap.tsx

```diff
@@ -0,0 +1,379 @@
+        <Typography component='h1'>Swap tokens</Typography>
+        <CardMedia image={settingIcon} className={classes.settingsIcon} onClick={handleClickSettings}/>
+        <Grid className={classes.slippage} >
+          <Slippage open={settings}
+            setSlippage={setSlippage}
+            handleClose={handleCloseSettings}
+            anchorEl={anchorEl}/>
+        </Grid>
+      </Grid>
+      <Grid container className={classes.root} direction='column'>
+        <Box className={classes.tokenComponentTextContainer}>
+          <Typography className={classes.tokenComponentText}>Est.: </Typography>
+          <Typography className={classes.tokenComponentText}>
+          Balance: {tokenFromIndex !== null
+              ? swap
+                ? printBN(tokens[tokenToIndex ?? 0].balance, tokens[tokenToIndex ?? 0].decimal)
+                : printBN(tokens[tokenFromIndex].balance, tokens[tokenFromIndex].decimal) : '0'}
+          </Typography>
+        </Box>
+        <ExchangeAmountInput
+          value={amountFrom}
+          className={classes.amountInput}
+          style={{ transform: swap !== null ? swap ? 'translateY(104px)' : 'translateY(0px)' : '' }}
+          setValue={value => {
+            if (value.match(/^\d*\.?\d*$/)) {
+              setAmountFrom(value)
+              updateEstimatedAmount(value)
+            }
+          }}
+          placeholder={'0.0'}
+          onMaxClick={() => {
+            if (tokenToIndex !== null && tokenFromIndex !== null) {
+              if (swap) {
+                setAmountTo(printBN(tokens[tokenToIndex].balance, tokens[tokenToIndex].decimal))
+                updateFromEstimatedAmount(
+                  printBN(tokens[tokenToIndex].balance, tokens[tokenToIndex].decimal)
+                )
+              } else {
+                setAmountFrom(printBN(tokens[tokenFromIndex].balance, tokens[tokenFromIndex].decimal))
+                updateEstimatedAmount(printBN(tokens[tokenFromIndex].balance, tokens[tokenFromIndex].decimal))
+              }
+            }
+          }}
+          tokens={swap ? tokenY.map(({ symbol, name, icon }) => ({
+            symbol,
+            name,
+            icon
+          }))
+            : tokens.map(({ symbol, name, icon }) => ({
+              symbol,
+              name,
+              icon
+            }))}
+          current={tokenFromIndex !== null ? tokens[tokenFromIndex].symbol : null}
+          onSelect={(chosen: number) => setTokenFromIndex(chosen)}
+        />
+        <Box className={classes.tokenComponentTextContainer}>
+          <Box className={classes.swapArrowBox}>
+            <CardMedia image={SwapArrows}
+              style={{ transform: swap !== null ? swap ? 'rotate(180deg)' : 'rotate(0deg)' : '' }}
+              className={classes.swapArrows} onClick={() => {
+                if (tokenToIndex !== null) {
+                  swap !== null ? setSwap(!swap) : setSwap(true)
+                } else {
```

Comment:
why is this case empty?

Reply:
maybe i want to put if statement only, i'll remove else